src/charge_controller/dev/LCDDisplay.cpp

```diff
@@ -32,38 +24,149 @@ void LCDDisplay::display() {
 
     lcd.setTextForSection(2, batteryTwoStatus);
 
-    char batteryOneVolt[32];
-    std::sprintf(batteryOneVolt, "%d V", batteryOneVoltage);
+    char batteryOneVolt[16] = {0};
+    std::sprintf(batteryOneVolt, "%d V", batteryMinVoltages[0]);
     lcd.setTextForSection(3, batteryOneVolt);
-    char percentage[32];
+
+    char percentage[16] = {0};
     std::sprintf(percentage, "%d.%02d", chargePercentage / 100, chargePercentage % 100);
     lcd.setTextForSection(4, percentage);
 
-    char batteryTwoVolt[32];
-    std::sprintf(batteryTwoVolt, "%d V", batteryTwoVoltage);
+    char batteryTwoVolt[16] = {0};
+    std::sprintf(batteryTwoVolt, "%d V", batteryMinVoltages[1]);
     lcd.setTextForSection(5, batteryTwoVolt);
 
-    char bat1MinTemp[32];
-    std::sprintf(bat1MinTemp, "%d C", batteryOneMinTemp);
+    char bat1MinTemp[16] = {0};
+    std::sprintf(bat1MinTemp, "%d C", batteryMinTemps[0]);
     lcd.setTextForSection(6, bat1MinTemp);
 
-    char chargerVoltage[32];
-    std::sprintf(chargerVoltage, "%d V", chargeControllerVoltage);
+    char chargerVoltage[16] = {0};
+    std::sprintf(chargerVoltage, "%d.%d V", chargeControllerVoltage / 10, chargeControllerVoltage % 10);
     lcd.setTextForSection(7, chargerVoltage);
 
-    char bat2MinTemp[32];
-    std::sprintf(bat2MinTemp, "%d C", batteryTwoMinTemp);
+    char bat2MinTemp[16] = {0};
+    std::sprintf(bat2MinTemp, "%d C", batteryMinTemps[1]);
     lcd.setTextForSection(8, bat2MinTemp);
 
-    char bat1MaxTemp[32];
-    std::sprintf(bat1MaxTemp, "%d C", batteryOneMaxTemp);
+    char bat1MaxTemp[16] = {0};
+    std::sprintf(bat1MaxTemp, "%d C", batteryMaxTemps[0]);
     lcd.setTextForSection(9, bat1MaxTemp);
 
-    char currentDisplay[32];
-    std::sprintf(currentDisplay, "%d A", chargeControllerCurrent);
+    char currentDisplay[16] = {0};
+    std::sprintf(currentDisplay, "%d.%d A", chargeControllerCurrent / 10, chargeControllerCurrent % 10);
     lcd.setTextForSection(10, currentDisplay);
 
-    char bat2MaxTemp[32];
-    std::sprintf(bat2MaxTemp, "%d C", batteryTwoMaxTemp);
+    char bat2MaxTemp[16] = {0};
+    std::sprintf(bat2MaxTemp, "%d C", batteryMaxTemps[1]);
     lcd.setTextForSection(11, bat2MaxTemp);
 }
+
+void LCDDisplay::setChargerVoltage(uint16_t voltage) {
+    chargeControllerVoltage = voltage;
+}
+
+void LCDDisplay::setChargerCurrent(uint16_t current) {
+    chargeControllerCurrent = current;
+}
+
+void LCDDisplay::setBatteryStatus(BMSManager::BMSStatus status, uint8_t index) {
+    if (index == 0) {
```

Comment:
Should probably at least use an array with two strings in it because that would be a little better